scripts/hal_alignment/test_hal_gab_access.pl

```diff
@@ -0,0 +1,268 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+use warnings;
+use strict;
+
+=head1 NAME
+
+test_hal_gab_access.pl
+
+=head1 DESCRIPTION
+
+This script tests access to genomic align blocks from an MLSS backed by a HAL file.
+Both access via a slices and pairwise access via dnafrags is tested.
+
+=head1 SYNOPSIS
+
+	test_hal_gab_access.pl --regions regions.tsv --mlss 312167 --out output.json
+
+=head1 OPTIONS
+
+=head2 GETTING HELP
+
+=over
+
+=item B<[--help]>
+
+Prints help message and exits.
+
+=back
+
+=head2 GENERAL CONFIGURATION
+
+=over
+
+=item B<[--reg_conf registry_configuration_file]>
+
+The Bio::EnsEMBL::Registry configuration file. If none given,
+the L<--compara> option must be a URL or the COMPARA_REG_PATH
+environmental variable must be set.
+
+=item B<[--compara compara_db_name_or_alias]>
+
+The compara database to use. You can use either the original name or any of the
+aliases given in the registry_configuration_file. DEFAULT VALUE: compara_curr
+(assumes the registry information is given).
+
+=item B<--mlss method_link_species_set_id>
+
+A MethodLinkSpeciesSet identifier.
+
+=item B<--regions regions_stv>
+
+A TSV containing the test regions. It can be generated by the B<sample_genomic_regions.pl> script.
+
+=item B<[--size region_size]>
+
+Size of the regions to sample.
+
+=item B<[--hal_dir hal_directory]>
+
+The path to the directory containing the HAL files. Can also be specified through the B<COMPARA_HAL_DIR>
+environmental variable.
+
+=item B<[--out output_json]>
+
+Name of the output JSON file.
+
+=back
+
+=cut
+
+use Getopt::Long;
+use Text::CSV qw(csv);
+use Proc::ProcessTable;
+use JSON;
+
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+
+my $help;
+my $reg_conf;
+my $compara = 'compara_curr';
+my $mlss_id;
+my $ss_id;
+my $size = 5000;
+my $nr = 3;
+my $regions_tsv;
+my $outfile = "test_hal_gab_access.json";
+my $hal_dir;
+
+GetOptions(
+    'help'          => \$help,
+    'reg_conf=s'    => \$reg_conf,
+    'compara=s'     => \$compara,
+    'mlss=i'        => \$mlss_id,
+    'regions=s'     => \$regions_tsv,
+    'hal_dir=s'     => \$hal_dir,
+    'out=s'         => \$outfile,
+);
+
+# Print Help and exit if help is requested
+if ($help) {
+    use Pod::Usage;
+    pod2usage({-exitvalue => 0, -verbose => 2});
+}
+
+# Process command line parameters:
+die("The MLSS ID must be specified!") if !$mlss_id;
+die("The regions TSV file must be specified!") if !$regions_tsv;
+
+if ($hal_dir && $ENV{COMPARA_HAL_DIR}) {
+    die("Cannot set HAL dir as the COMPARA_HAL_DIR envirnomental variable is already set!");
+} elsif ($hal_dir) {
+    $ENV{COMPARA_HAL_DIR} = $hal_dir;
+}
+
+$reg_conf = $ENV{COMPARA_REG_PATH} if !$reg_conf;
+
+use Data::Dumper;
+
```

Comment:
Unused import:
```suggestion
use Data::Dumper;

```

Reply:
For some reason I cannot apply this suggestion. Will delete it in a commit.